### PR TITLE
Feature/authentication

### LINKED ISSUE #7: Authentication
User authentication and admin authentication

REQUIREMENTS:

To sign up:
Username - it must be a unique
Email - it must be a unique
Password
after a user registration, user must be confirm account via email.

allow users to Sign in and sign out.

Add profile settings page

Acceptance Criteria

 Sign up using username, email and password.
 Sign in allow user to initiate sesion using email and password
 Sign out allow use to end session
 Users must be confirm account via email.
 Allow user to edit profile : username, email, password

### PR DESCRIPTION
**Pull request by Brayan9105**
Implement the authentication proceses to users and admins

Implement Devise to manage the authentication of the users( sign up, sign in and sign out), when a user sign up in the app, this must be confirm the account via email, in the case of the admin, implement ActiveAdmin. the users can edit their profiles

Gems:
- Devise
- ActiveAdmin

To run the test file:
- type 'rspec' in the console

to run the seeds:
- type 'rails db:seed' in the console

### CODE REVIEW

**Review by Martin11-RL — APPROVED** [review]

**Review by Magic-Kowel — APPROVED** [review]

**Review by Magic-Kowel — APPROVED** [review]

**Review by Magic-Kowel — APPROVED** [review]

**Review by Magic-Kowel — APPROVED** [review]

File: app/helpers/dashboard_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module DashboardHelper
```
Comment (ccarruitero) [inline app/helpers/dashboard_helper.rb:1]: This file is unused, if not plan to use please remove

File: app/views/devise/mailer/reset_password_instructions.html.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<p>Hello <%= @resource.email %>!</p>
```
Comment (ccarruitero) [inline app/views/devise/mailer/reset_password_instructions.html.erb:1]: You could localize the mail content

File: app/views/devise/passwords/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<h2>Change your password</h2>
```
Comment (ccarruitero) [inline app/views/devise/passwords/edit.html.erb:1]: You could localize your views too

**Review by ccarruitero — APPROVED** [review]
Seems good, just take take care of not add unused files and try to localize all your views

**Review by fedgut — APPROVED** [review]

**Review by brayandiazc — COMMENTED** [review]
ok

**Pull request merged**